node/external/transactionAPI/gasUsedAndFeeProcessor.go

```diff
@@ -27,11 +42,35 @@ func (gfp *gasUsedAndFeeProcessor) computeAndAttachGasUsedAndFee(tx *transaction
 	tx.GasUsed = gasUsed
 	tx.Fee = fee.String()
 
-	if tx.IsRelayed || gfp.isESDTOperationWithSCCall(tx) {
+	if gfp.isESDTOperationWithSCCall(tx) {
 		tx.GasUsed = tx.GasLimit
 		tx.Fee = tx.InitiallyPaidFee
 	}
 
+	if gfp.isGuardianOperation(tx) {
```

Comment:
Maybe add a comment to explain why we handle these builtin functions differently than other builtin functions (in this regard).